python/tk_multi_publish2/api/plugins/instance_base.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) 2017 Shotgun Software Inc.
```

Comment:
Pretty close to https://github.com/shotgunsoftware/tk-multi-publish2/blob/master/python/tk_multi_publish2/processing/plugin.py#L19 with the added `id` property and the `__getstate__`, `__setstate__` methods (which I haven't documented yet... I'll do this asap).